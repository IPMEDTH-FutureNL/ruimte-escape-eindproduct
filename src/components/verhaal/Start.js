import React, { useState } from 'react';
import { useNavigate } from "react-router-dom";
import '../../css/start.css';
import Note from "../../img/gameplayImages/snippets/note.png"
import Astronaut from "../../img/gameplayImages/characters/character2_standard_schuin_mouthopen.svg"
import { FaInfoCircle } from "react-icons/fa";
import { FaAngleRight } from "react-icons/fa";
import TextCloud from '../../img/gameplayImages/snippets/textcloud.png'
// import IntroSound from '../../sound/Intro.mp3';


const Start = () => {
    // const audio = new Audio(IntroSound);
    const [text, setText] = useState("");
    const [playerState, setPlayerState] = useState("false");
    const [count, setCount] = useState(0);
    const navigate = useNavigate();
    
    const showPlayerForm = (number) => {
        let formOne = document.getElementById("container__eenSpeler");
        let formTwo = document.getElementById("container__tweeSpelers");
        if(number === 1){
            setPlayerState("false");
            formOne.style.display = "block";
            formTwo.style.display = "none";
        }else{
            setPlayerState("true");
            formOne.style.display = "none";
            formTwo.style.display = "block";
        }

    }
    
    const players = (e) => {
        e.preventDefault();
        // audio.play();
        let players = getPlayerNames();
        let introduction = document.getElementById("container__introduction");
        let gameplay = document.getElementById("container__gameplay");


        let playerOne = "";
        let playerTwo = "";

        playerOne = players[0];
        playerTwo = players[1];
        localStorage.setItem("playerOne", playerOne);
        localStorage.setItem("playerTwo", playerTwo);

        introduction.style.display = "none";
        gameplay.style.display = "block";
        localStorage.setItem('punten', 30);
  
        if(localStorage.getItem("playerTwo") === "undefined"){
            localStorage.setItem("playerTwo", "");
            setText("Welkom " + localStorage.getItem("playerOne") + " op het ruimteschip! Ik ben Max een van de astronauten op dit schip.")
        }else{
            setText("Welkom " + localStorage.getItem("playerOne") + " en " + localStorage.getItem("playerTwo") + " op het ruimteschip! Ik ben Max een van de astronauten op dit schip.")
        }
       
    }

    const getPlayerNames = () => {

        let input = "";

        if(playerState === "false"){
            input = document.getElementsByClassName("player__form-name-speler");
        }else{
            input = document.getElementsByClassName("player__form-name");
        }
        let players = [];
        let currentPlayer = ""
       

        for(let i = 0; i < input.length; i++){
            currentPlayer = input[i].value;
            players.push(currentPlayer);
        }

        return players;
    }

    const story = (newText) => {
        setCount(count + newText);

        switch(count){
            case 0:
                setText("We zijn bijna klaar om koers te zetten naar Mars!")
                break;
            case 1:
                if(localStorage.getItem("playerTwo") != null){
                    setText("We hebben alleen nog een probleem, de kaart klopt niet. Ik heb je hulp nodig!");
                }else{
                    setText("We hebben alleen nog een probleem, de kaart klopt niet. Ik heb jullie hulp nodig!");
                }
                break;
            case 2:
                navigate('/introPuzzle');
                break;
        }
    }

    return(
        <section className="container">
            <section className="containter__start">
                <section className="container__introduction" id="container__introduction">
                    <img src={Note} alt="note with rules" className="escaperoom__rules-note" draggable="false"></img>
                    <div className="escaperoom__rules">
                        <h2>Welkom bij de missie naar Mars!</h2>
                        <p>Voordat we beginnen nog een paar korte tips!</p>
                        <div className="rule">
                            <button className='rule-hint'> Tips</button>
                            <p className="hint-text">De tip knop, hier kan je tips krijgen over de puzzels de kosten alleen 10 punten!</p>
                        </div>
                        <div className="rule">
                            <div className="rule-clickable" />
                            De groene rondjes in de escaperoom, betekenen dat je erop kan klikken!
                        </div>
                        <div className="rule">
                            Met hoeveel personen zijn jullie?
                        </div>
                        <div className="button__group">
                            <button className="button primary" onClick={() => showPlayerForm(1)}>1</button>
                            <button className="button primary" onClick={() => showPlayerForm(2)}>2</button>
                        </div>
                    </div>


                    <div className="container__eenSpeler" id="container__eenSpeler">
                        <form className="player__form first" onSubmit={(e) => players(e)}>
                            <div className="inputdiv">
                                <label className="player__form-label">Naam</label>
                                <input className="player__form-name-speler"></input>
                            </div>
                            <input type="submit" value="Verder" className='button primary submit'/>
                        </form>
                    </div>

                    <div className="container__tweeSpelers" id="container__tweeSpelers">
                        <form className="player__form second" onSubmit={(e) => players(e)}>
                            <div className="inputdiv">
                                <label className="player__form-label">Naam</label>
                                <input className="player__form-name"></input>
                            </div>
                            <div className="inputdiv">
                                <label className="player__form-label">Naam</label>
                                <input className="player__form-name"></input>
                            </div>
                            <div className="inputdiv">
                                <input type="submit" value="Verder" className='button primary submitSecond'/>
                            </div>
                        </form>
                    </div>
                </section>

                <section className="container__gameplay" id="container__gameplay">
                    <div className="container__astronaut">
                        <div className="astronaut">
                            <img src={Astronaut} alt="astronaut" className="astronaut__image" draggable="false"></img>
                        </div>
                        <div className="textcloud">
                            <img src={TextCloud} alt="tekst balloon" className="textcloud__image" draggable="false"></img>
                            <div className="textcloud__field">
                                <p className="textcloud__text">{text}</p>
                                <button className="continue" onClick={() => story(1)}>Verder</button>
                            </div>
                        </div>
                    </div>
                </section>
            </section>
        </section>
    )
}

export default Start;