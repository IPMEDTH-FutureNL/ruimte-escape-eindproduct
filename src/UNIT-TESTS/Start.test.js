import { render , screen, fireEvent } from "@testing-library/react"
import {BrowserRouter as Router} from 'react-router-dom';
import Start from "../components/verhaal/Start";

test('on initial render of start, check if component is loaded', () => {
    const { asFragment }  = render(
        <Router>
            <Start></Start>
        </Router>
    )

    screen.debug();
    expect(asFragment()).toMatchSnapshot();
})

test('if change story could be clicked', () => {
    render(
        <Router>
            <Start></Start>
        </Router>
    )

    const btnStory = screen.getByTestId("story");
    fireEvent.click(btnStory);

    expect(fireEvent.click(btnStory)).toMatchSnapshot();
})

test('if on submit for player name works', () => {
    render(
        <Router>
            <Start></Start>
        </Router>
    )

    const submit = screen.getByTestId('onsubmit');
    fireEvent.click(submit);

    expect(fireEvent.click(submit)).toMatchSnapshot();
})