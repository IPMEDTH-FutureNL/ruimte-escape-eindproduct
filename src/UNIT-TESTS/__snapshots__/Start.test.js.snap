// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if change story could be clicked 1`] = `true`;

exports[`if on submit for player name works 1`] = `true`;

exports[`on initial render of start, check if component is loaded 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <section
      class="containter__start"
    >
      <section
        class="container__introduction"
        id="container__introduction"
      >
        <img
          alt="note with rules"
          class="escaperoom__rules-note"
          draggable="false"
          src="note.png"
        />
        <div
          class="escaperoom__rules"
        >
          <h2>
            Welkom bij de missie naar Mars!
          </h2>
          <p>
            Voordat we beginnen nog een paar korte tips!
          </p>
          <div
            class="rule"
          >
            <button
              class="rule-hint"
            >
               Tips
            </button>
            <p
              class="hint-text"
            >
              De tip knop, hier kan je tips krijgen over de puzzels de kosten alleen 10 punten!
            </p>
          </div>
          <div
            class="rule"
          >
            <div
              class="rule-clickable"
            />
            De groene rondjes in de escaperoom, betekenen dat je erop kan klikken!
          </div>
          <div
            class="rule"
          >
            Vul snel jullie namen in dan kunnen we beginnen!
          </div>
        </div>
        <form
          class="player__form"
          data-testid="onsubmit"
        >
          <div
            class="inputdiv"
          >
            <label
              class="player__form-label"
            >
              Naam
            </label>
            <input
              class="player__form-name"
            />
          </div>
          <div
            class="inputdiv"
          >
            <label
              class="player__form-label"
            >
              Naam
            </label>
            <input
              class="player__form-name"
            />
          </div>
          <div
            class="inputdiv"
          >
            <input
              class="player__form-submit"
              type="submit"
              value="Verder"
            />
          </div>
        </form>
      </section>
      <section
        class="container__gameplay"
        id="container__gameplay"
      >
        <div
          class="container__astronaut"
        >
          <div
            class="astronaut"
          >
            <img
              alt="astronaut"
              class="astronaut__image"
              draggable="false"
              src="character2_standard_schuin_mouthopen.svg"
            />
          </div>
          <div
            class="textcloud"
          >
            <img
              alt="tekst balloon"
              class="textcloud__image"
              draggable="false"
              src="textcloud.png"
            />
            <div
              class="textcloud__field"
            >
              <p
                class="textcloud__text"
              />
              <button
                class="continue"
                data-testid="story"
              >
                Verder
              </button>
            </div>
          </div>
        </div>
      </section>
    </section>
  </section>
</DocumentFragment>
`;
